### ISSUE TITLE
Database does not contain all the source files

### ISSUE THREAD

**Issue by fossilet** [github/codeql#13875]
**Database does not contain all the Java files**

I am using codeql to create database for [webgoat/webgoat](https://github.com/webgoat/webgoat). Either using autobuild, or specifying command with `--command='./mvnw clean install'`, the source archive does not contain all the relevant Java files, e.g. `src/main/java/org/owasp/webgoat/container/WebSecurityConfig.java` is missing.

Besides, codeql cannot find the CSRF disabled problem. I suspect missing this file is the cause of this.

Is this supposed behavior or am I missing some configuration? Thanks.

**Comment by smowton** [comment]
This happens because WebGoat uses Lombok, which currently isn't supported -- you'd need to use `delombok` or the Delombok Maven plugin to produce a plain-Java project that CodeQL can analyse in full.

However! We're currently testing a more complete solution. If you set `CODEQL_EXTRACTOR_JAVA_RUN_ANNOTATION_PROCESSORS=true` in your environment, you should find all files are extracted. Just beware that because this is a feature in testing there may well be bugs we haven't ironed out yet ahead of the  feature getting switched on by default. If you find anything wrong with the feature please let us know, but be aware that while it's in beta we'll make a best effort to resolve any snags, but no promises.

**Comment by fossilet** [comment]
I've tried delombok. From the help from awhitford/lombok.maven/issues/17, It generates non-annotated sources and they are included in the database.

I tried the new environment. It is much easy to use, and it generated more results than delombok for the same query suite.
After delombok, I get 79 locations in 197 results; the latter gets 82 locations in 202 results.

I will use the environment thus far. Thanks.
